[Docs] Expand introduction to Ray in Multi-node deployment section (#21584)
vllm-project/vllm: docs/serving/distributed_serving.md

## Multi-node deployment

If a single node lacks sufficient GPUs to hold the model, deploy vLLM across multiple nodes. Ensure that every node provides an identical execution environment, including the model path and Python packages. Using container images is recommended because they provide a convenient way to keep environments consistent and to hide host heterogeneity.

### What is Ray?

> Ray is a distributed computing framework that makes your code scale-agnostic. Multi-node vLLM deployments require Ray as the runtime engine.

Reviewer: ```suggestion
Ray is a distributed computing framework for scaling python programs. Multi-node vLLM deployments require Ray as the runtime engine.
```
